enclave/core/sgx/init.c

```diff
@@ -61,7 +69,12 @@ static oe_result_t _eeid_patch_memory()
         uint8_t* enclave_base = (uint8_t*)__oe_get_enclave_base();
         uint8_t* heap_base = (uint8_t*)__oe_get_heap_base();
         oe_eeid_marker_t* marker = (oe_eeid_marker_t*)heap_base;
-        oe_eeid = (oe_eeid_t*)(enclave_base + marker->offset);
+        uint8_t* eeid_ptr = enclave_base + marker->offset;
+
+        if (!_is_within_eeid_enclave(eeid_ptr))
```

Comment:
Need to check the fixed size portion of eeid (sizeof(oe_eeid_t)) is within the enclave first,  then read the size of the variable length portion (eeid_ptr->data_size + eeid_ptr->signature_size), and check the whole data structure is within the enclave.

Reply:
Good point, we need to check the end as well.

Reply:
(Done)